oibackoff/index.d.ts

```diff
@@ -0,0 +1,46 @@
+// Type definitions for oibackoff
+// Project: https://github.com/chilts/oibackoff
+// Definitions by: Joshua DeVinney <https://github.com/geoffreak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export declare function backoff(opts: BackoffOptions): BackoffInstance;
+
+export declare interface BackoffOptions {
+    maxTries?: number;
+    algorithm?: 'incremental' | 'exponential' | 'fibonacci';
+    delayRatio?: number;
+    maxDelay?: number;
+}
+
+export declare interface BackoffInstance {
+    (fn: Function, callback: Function): void;
```

Comment:
Alright! Sometimes the simplest solutions are the best.
[Version 3 script that generates <85 lines but limits callbacks to 2 arguments](https://gist.github.com/geoffreak/8ac60f3f995f56401c862bc58b447817)